NitroxModel/DataStructures/ThreadSafeDictionary.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using ProtoBufNet;
+
+namespace NitroxModel.DataStructures
+{
+    [DebuggerDisplay("Items = {" + nameof(Entries) + "}")]
+    [ProtoContract]
```

Comment:
Actually not sure how it serializes. Maybe Protobuf uses the properties for it to get/set data. I know that it saves correctly though 😅

Reply:
It might be because it extends IDictionary. 

For the future, I don't think we want this to be protobug enabled because it has notoriously bad handling of dictionaries.  There should only be 1 or 2 places that actually persist a dictionary (such as PDA which hasn't been migrated yet).  We have moved everything else over to store only lists and have the indexing as an implementation feature of the class that needs fast lookups.  For now we can leave it.